mlir/test/CMakeLists.txt

```diff
@@ -114,10 +115,12 @@ endif()
 
 # Enable the ROCK E2E tests if -DROCK_E2E_TEST_ENABLED=1
 if(ROCK_E2E_TEST_ENABLED OR ROCMLIR_DRIVER_PR_E2E_TEST_ENABLED)
-  list(APPEND ROCMLIR_TEST_DEPENDS e2e_tests)
+  list(APPEND ROCMLIR_TEST_DEPENDS e2e_tests fusion-e2e-tests)
   add_subdirectory(e2e)
+  add_subdirectory(fusion)
```

Comment:
I'm not liking this. There are non e2e fusion tests.

Could you instead refactor the tests from `fusion/e2e` to `e2e/fusion`

Reply:
There are also non-fusion e2e tests.  I was leaving the original arrangement, but I'll consider moving them.